### ISSUE TITLE
Expose ML implementation name in the REPL

### ISSUE THREAD

**Issue by eduardoleon** [SMLFamily/BasisLibrary#15]
I'd like interactive Successor ML implementations to expose their name as a standardized global constant:

```sml
val topLevelName : unit -> string
```

This way, it would be easier to work around differences between implementations, writing code that looks like this:

```sml
case topLevelName () of
    "Poly/ML" => use "Foo.PolyML.sml"
  | "MosML" => use "Foo.MosML.sml"
  | "SML/NJ" => use "Foo.SMLNJ.sml"
```

Thus avoiding the need to use arcane external tools for building programs.

**Comment by JohnReppy** [comment]
I do not think that we should be relying on the REPL for configuring builds.  I think that something like  [smackage](https://github.com/standardml/smackage) is a better approach.

That said, adding a standard way to query vendor information from SML does seem like a reasonable feature.  It could either go into a new module or be added to the `General` structure.
